### ISSUE TITLE
CiviCRM_API3_Exception: DB Constraint Violation

### ISSUE THREAD

**Issue by laryn** [civicrm/civicrm-backdrop#146]
I'm not sure why, but I'm getting this fatal error on a particular site after recent upgrades to CivICRM:

> CiviCRM_API3_Exception: DB Constraint Violation - region_id should possibly be marked as mandatory for Country,get API. If so, please raise a bug report. in civicrm_api3() (line 134 of /app/modules/civicrm/api/api.php).

Strangely, it seems to be originating in `civicrm.install` where the CiviCRM version is checked and reported on the Backdrop Status Report page. More strangely, I've only noticed it on one site so far (even on the same server as another site without the problem).

If I remove the `$vc->fetch();` line, it seems to work fine on the sites I've tested it on. Does that seem reasonable?

**Comment by herbdool** [comment]
@laryn I think we should get it closer to civicrm-drupal, which has a simpler install file and isn't checking the version. Though I'm a bit confused by https://github.com/civicrm/civicrm-drupal/commit/c3f300aeaf10aa70a1e1e97bed6d85a8972fd4b5 which seems to introduce `civicrm_requirements()` where I thought there would be a commit which simplified it.

**Comment by herbdool** [comment]
Or there may be some value in keeping the version check. Would be good to see if there's a better way to check the version.

**Comment by herbdool** [comment]
I like git blame! Looks like you @laryn added the version check! https://github.com/civicrm/civicrm-backdrop/issues/62

**Comment by laryn** [comment]
Yes, I accept responsibility but not sure why it stopped working :sweat_smile: This was the bit that adds a CiviCRM line on the Backdrop status report (and flags the red warning button in the admin toolbar if there's a CiviCRM security update).

**Comment by laryn** [comment]
I'm not seeing this on any other sites so I'm going to close for now. Will reopen if I can provide STR or more information to work from.

**Comment by herbdool** [comment]
I think we can change `$vc->fetch()` to `$vc->initialize()`. The latter will first read from the cache file before attempting to fetch from remote. That might make it more robust.

**Comment by herbdool** [comment]
I've updated your PR @laryn. I can see why even just removing `$vc-fetch()` worked since it would fallback to the just displaying the localVersion if it didn't have more versionInfo. I also added ability to show any new patches even if not critical.

**Comment by laryn** [comment]
@herbdool Looks good -- I made a slight tweak to your changes in the PR to reduce code volume/duplication slightly.

**Comment by herbdool** [comment]
merged! Thanks @laryn.
